2-TokenCache/Console-TokenCache/Program.cs

```diff
@@ -0,0 +1,218 @@
+                     CacheSettings.LinuxKeyRingAttr1,
+                     CacheSettings.LinuxKeyRingAttr2)
+                 .WithMacKeyChain(
+                     CacheSettings.KeyChainServiceName,
+                     CacheSettings.KeyChainAccountName)
+                 .Build();
+
+            // This hooks up the cross-platform cache into MSAL
+            var cacheHelper = await MsalCacheHelper.CreateAsync(storageProperties);
+            cacheHelper.RegisterCache(app.UserTokenCache);
+
+            // Subscribing to the CacheChanged event
+            cacheHelper.CacheChanged += (object sender, CacheChangedEventArgs eventArgs) =>
+            {
+                Console.ForegroundColor = ConsoleColor.DarkYellow;
+                Console.WriteLine($"Cache Changed, Added: {eventArgs.AccountsAdded.Count()} Removed: {eventArgs.AccountsRemoved.Count()}");
+                Console.ResetColor();
+            };
+
+            string[] scopes = new[] { "user.read" };
+            AuthenticationResult result;
+
+            try
+            {
+                var accounts = await app.GetAccountsAsync();
+
+                // Try to acquire an access token from the cache. If an interaction is required, 
+                // MsalUiRequiredException will be thrown.
+                result = await app.AcquireTokenSilent(scopes, accounts.FirstOrDefault())
+                            .ExecuteAsync();
+            }
+            catch (MsalUiRequiredException)
+            {
+                // Acquiring an access token interactively. MSAL will cache it so we can use AcquireTokenSilent
+                // on future calls.
+                result = await app.AcquireTokenInteractive(scopes)
+                            .ExecuteAsync();
+            }
+
+            string graphApiUrl = configuration.GetValue<string>("GraphApiUrl");
+            // Instantiating GraphServiceClient and using the access token acquired above.
+            var graphClient = GetGraphServiceClient(result.AccessToken, graphApiUrl);
+
+            // Calling the /me endpoint
+            var me = await graphClient.Me.Request().GetAsync();
+
+            // Printing the results
+            DisplayGraphResult(result, me);
+
+            while (true)
+            {
+                // Display menu
+                Console.WriteLine("------------ MENU ------------");
+                Console.WriteLine("1. Acquire Token Interactive");
+                Console.WriteLine("2. Acquire Token Silent");
+                Console.WriteLine("3. Display Accounts (reads the cache)");
+                Console.WriteLine("4. Clear cache");
+                Console.WriteLine("x. Exit app");
+                Console.Write("Enter your Selection:");
+                char.TryParse(Console.ReadLine(), out var selection);
+
+                try
+                {
+                    switch (selection)
+                    {
+                        case '1': // Interactive
```

Comment:
I like this menu of options, but I wonder if it wouldn't be better to showcase our "silent + interactive" pattern instead of having separate "silent" / "interactive" options? People who are not familiar with MSAL will not understand the purpose of "silent"...

Reply:
Actually, I agree. I had a hard time with it when playing with the app (but did not realize it)

Reply:
I have combined the menu options